src/main/java/com/yelp/nrtsearch/server/luceneserver/state/BackendGlobalState.java

```diff
@@ -177,6 +177,29 @@ StateBackend getStateBackend() {
     return stateBackend;
   }
 
+  @Override
+  public synchronized void reloadStateFromBackend() throws IOException {
+    GlobalStateInfo newGlobalStateInfo = getStateBackend().loadOrCreateGlobalState();
+    Map<String, IndexStateManager> newManagerMap = new HashMap<>();
+    for (Map.Entry<String, IndexGlobalState> entry :
+        newGlobalStateInfo.getIndicesMap().entrySet()) {
+      String indexName = entry.getKey();
+      IndexStateManager stateManager;
+      if (immutableState.indexStateManagerMap.containsKey(indexName)) {
+        stateManager = immutableState.indexStateManagerMap.get(indexName);
```

Comment:
It might be good to check if the previous state manager also has the same id as the index entry, otherwise create a new one. You might need to add a method to expose this.